src/test/java/com/fasterxml/jackson/databind/deser/TestCustomSubTypeValidator.java

```diff
@@ -0,0 +1,277 @@
+package com.fasterxml.jackson.databind.deser;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.cfg.DeserializerFactoryConfig;
+import com.fasterxml.jackson.databind.exc.InvalidDefinitionException;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.jsontype.impl.SubTypeValidator;
+
+public class TestCustomSubTypeValidator 
+    extends BaseMapTest
+{
+    /*
+    /**********************************************************
+    /* Some sample documents:
+    /**********************************************************
+     */
+
+    protected static final String STRING_VALUE_JSON = "\"just a plain JSON string\"";
+
+
+    protected static final String ZOO_JSON = "{"
+            + "  \"name\" : \"Jacksonville Zoo\","
+            + "  \"animals\": ["
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Warthog\","
+            + "      \"name\" : \"Pumba\""
+            + "    },"
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Lion\","
+            + "      \"name\" : \"Kimba\""
+            + "    },"
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Goat\","
+            + "      \"name\" : \"Goaty McGoatface\""
+            + "    },"
+            + "    {"
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Penguin\","
+            + "      \"name\" : \"Tux\""
+            + "    }"
+            + "  ]"
+            + "}";
+
+    protected static final String PETTING_ZOO_JSON = "{"
+            + "  \"name\" : \"Jacksonville Petting Zoo\","
+            + "  \"animals\": ["
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Goat\","
+            + "      \"name\" : \"Bill\""
+            + "    },"
+            + "    { "
+            + "      \"species\" : \"com.fasterxml.jackson.databind.deser.TestCustomSubTypeValidator$Sheep\","
+            + "      \"name\" : \"Shaun\""
+            + "    }"
+            + "  ]"
+            + "}";
+
+    /*
+    /**********************************************************
+    /* Helper classes (beans)
```

Comment:
I have created helper beans for this test from scratch.
Could I reuse existing beans that make use of `@JsonTypeInfo`? Where to find these?